src/plugins/intel_npu/src/plugin/npuw/compiled_model.hpp

```diff
@@ -212,7 +222,11 @@ class CompiledModel : public ov::npuw::ICompiledModel {
         execution_stats stat;
 
         void serialize(std::ostream& stream, const ov::npuw::s11n::WeightsContext& ctx) const;
-        void deserialize(std::istream& stream, const ov::npuw::s11n::WeightsContext& ctx);
+        void deserialize(std::istream& stream,
```

Comment:
Could you please wrap it under `PyramidCtx` or something? So it's easily extended in the future and simplifies the signature. E.g. other context we have https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_npu/src/plugin/npuw/serialization.hpp#L96